src/app/components/EditorInterface.tsx

```diff
@@ -2,27 +2,123 @@
 
 import { When } from '@/components/hoc/When'
 import { useAppState } from '@/hooks/useAppState'
+import { useEditor, EditorContent } from '@tiptap/react'
+
+import Document from '@tiptap/extension-document'
+import Paragraph from '@tiptap/extension-paragraph'
+import Heading from '@tiptap/extension-heading'
+import Text from '@tiptap/extension-text'
+import OrderedList from '@tiptap/extension-ordered-list'
+import BulletList from '@tiptap/extension-bullet-list'
+import ListItem from '@tiptap/extension-list-item'
+import Image from '@tiptap/extension-image'
+import Table from '@tiptap/extension-table'
+import TableCell from '@tiptap/extension-table-cell'
+import TableHeader from '@tiptap/extension-table-header'
+import TableRow from '@tiptap/extension-table-row'
+import Underline from '@tiptap/extension-underline'
+import CodeBlock from '@tiptap/extension-code-block'
+import Code from '@tiptap/extension-code'
+import Link from '@tiptap/extension-link'
+import CalloutExtension from '@/components/tiptap/callout/CalloutExtension'
+import Bold from '@tiptap/extension-bold'
+import Italic from '@tiptap/extension-italic'
+import Strike from '@tiptap/extension-strike'
+import Gapcursor from '@tiptap/extension-gapcursor'
+
+import LinkInput from '@/components/tiptap/linkInput/LinkInput'
+import FloatingMenuContainer from '@/components/tiptap/floatingMenu/FloatingMenu'
+import BubbleMenuContainer from '@/components/tiptap/bubbleMenu/BubbleMenu'
+import { useEffect } from 'react'
+import AutofieldSelector from '@/components/tiptap/autofieldSelector/AutofieldSelector'
 
 const EditorInterface = () => {
   const appState = useAppState()
 
+  const editor = useEditor({
+    extensions: [
```

Comment:
may be we can define this configuration outside of the component and memoize or use useRef